ui-guide/Components/FormControl.purs

```diff
@@ -21,40 +20,34 @@ type State =
   { formPanelIsOpen :: Boolean }
 
 data Query a
-  = NoOp a
-  | HandleFormHeaderClick MouseEvent a
-  | ToggleFormPanel MouseEvent a
+data Action
+  = HandleFormHeaderClick MouseEvent
+  | ToggleFormPanel MouseEvent
 
 type Input = Unit
 
 type Message = Void
 
 component :: H.Component HH.HTML Query Input Message Aff
 component =
-  H.component
+  H.mkComponent
     { initialState: const { formPanelIsOpen: false }
     , render
-    , eval
-    , receiver: const Nothing
+    , eval: H.mkEval $ H.defaultEval { handleAction = handleAction }
     }
   where
-    eval :: Query ~> H.ComponentDSL State Query Message Aff
-    eval = case _ of
-      NoOp a -> do
-        pure a
-
-      HandleFormHeaderClick _ a -> do
+    handleAction :: Action -> H.HalogenM State Action () Message Aff Unit
+    handleAction = case _ of
+      HandleFormHeaderClick _ -> do
         H.liftEffect (log "submit form")
-        pure a
 
-      ToggleFormPanel _ a -> do
+      ToggleFormPanel _ -> do
         state <- H.get
         H.modify_ (_ { formPanelIsOpen = not state.formPanelIsOpen })
-        pure a
 
-    render :: State -> H.ComponentHTML Query
+    render :: State -> H.ComponentHTML Action () Aff
     render state =
-      let css :: ∀ p i. String -> H.IProp ( "class" :: String | p ) i
+      let css :: ∀ p i. String -> HH.IProp ( "class" :: String | p ) i
           css = HP.class_ <<< HH.ClassName
```

Comment:
Not that this needs to be part of this PR, but this should prob just be `Ocelot.HTML.Properties.css`